hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,18 @@
+'use client'
+import { useState, useEffect } from 'react'
+
+export const useMediaQuery = (query) => {
+  const [matches, setMatches] = useState(false)
+
+  useEffect(() => {
+    const media = window.matchMedia(query)
```

Comment:
Nitpick but I think some blank lines here would help improve readability a little, something like:

```ts
useEffect(() => {
    const media = window.matchMedia(query)
    
    if (media.matches !== matches) {
      setMatches(media.matches)
    }
    
    const listener = () => setMatches(media.matches)
    
    window.addEventListener('resize', listener)
    return () => window.removeEventListener('resize', listener)
  }, [matches, query])
``

Reply:
Addressed in 9b4c5743cb1375d2eae011ab956848b2f92682c6

It does improve readability!